infrastructure/container/container.go

```diff
@@ -0,0 +1,331 @@
+	}
+
+	if fileInfo.IsDir() {
+		// Container scanner operates on file basis, not directories
+		return []types.Issue{}, nil
+	}
+
+	// Only scan Dockerfile and docker-compose files
+	fileName := strings.ToLower(fileInfo.Name())
+	if !sc.isDockerFile(fileName) {
+		return []types.Issue{}, nil
+	}
+
+	bytes, err := os.ReadFile(string(path))
+	if err != nil {
+		sc.errorReporter.CaptureError(err)
+		log.Err(err).Str("method", "container.Scan").Msg("Error while reading file")
+		return []types.Issue{}, err
+	}
+
+	return sc.scanDockerFile(ctx, string(path), string(bytes))
+}
+
+func (sc *Scanner) isDockerFile(fileName string) bool {
+	lowerFileName := strings.ToLower(fileName)
+	return lowerFileName == "dockerfile" ||
+		strings.HasPrefix(lowerFileName, "dockerfile.") ||
+		lowerFileName == "docker-compose.yml" ||
+		lowerFileName == "docker-compose.yaml"
+}
+
+func (sc *Scanner) scanDockerFile(ctx context.Context, filePath string, content string) ([]types.Issue, error) {
+	issues := make([]types.Issue, 0)
+
+	// Regex to match FROM instructions in Dockerfile
+	fromRegex := regexp.MustCompile(`(?i)^\s*FROM\s+([^\s]+)`)
+
+	lines := strings.Split(strings.ReplaceAll(content, "\r", ""), "\n")
+	for lineNum, line := range lines {
+		matches := fromRegex.FindStringSubmatch(line)
+		if matches == nil {
+			continue
+		}
+
+		baseImage := matches[1]
+		// Skip scratch and multi-stage build references
+		if baseImage == "scratch" || strings.Contains(baseImage, " as ") {
+			continue
+		}
+
+		// Get recommendation from API
+		recommendation, err := sc.apiClient.GetBaseImageRecommendation(ctx, baseImage)
+		if err != nil {
+			sc.errorReporter.CaptureError(err)
+			log.Err(err).Str("method", "container.scanDockerFile").Msg("Error getting base image recommendation")
+			continue
+		}
+
+		// Create issue with code action to replace base image
+		issue := sc.createBaseImageIssue(filePath, lineNum, line, baseImage, recommendation)
+		issues = append(issues, issue)
+	}
+
+	return issues, nil
+}
+
+func (sc *Scanner) createBaseImageIssue(filePath string, lineNum int, line string, baseImage string, recommendation *BaseImageRecommendation) types.Issue {
+	// Find the position of the base image in the line
+	fromIndex := strings.Index(strings.ToLower(line), "from")
+	imageStart := fromIndex + 4 // "FROM" length
+	for imageStart < len(line) && line[imageStart] == ' ' {
+		imageStart++ // Skip whitespace
+	}
+	imageEnd := imageStart + len(baseImage)
+
+	r := types.Range{
+		Start: types.Position{Line: lineNum, Character: imageStart},
+		End:   types.Position{Line: lineNum, Character: imageEnd},
+	}
+
+	// Create text edit for code action
+	textEdit := types.TextEdit{
+		Range:   r,
+		NewText: recommendation.RecommendedImage,
+	}
+
+	workspaceEdit := &types.WorkspaceEdit{
+		Changes: map[string][]types.TextEdit{
+			filePath: {textEdit},
+		},
+	}
+
+	codeAction, err := snyk.NewCodeAction(
+		fmt.Sprintf("Upgrade to %s", recommendation.RecommendedImage),
```

Comment:
We may want to add ` (Snyk)` at the end of the action title.

Reply:
Also, in Open Source, we add the number of issues that would be fixed, and the number of issues that would remain in the text. Having that for container, would be great!